yarn-project/end-to-end/src/e2e_p2p/gossip_network_no_cheat.test.ts

```diff
@@ -0,0 +1,215 @@
+    expect(attesters.length).toBe(NUM_NODES);
+
+    // Check that the validators are added correctly
+    const withdrawer = await stakingAssetHandler.read.withdrawer();
+    for (const validator of validators) {
+      const info = await rollup.read.getInfo([validator.attester]);
+      expect(info.proposer).toBe(validator.proposer);
+      expect(info.withdrawer).toBe(withdrawer);
+    }
+
+    const slotsInEpoch = await rollup.read.getEpochDuration();
+    const timestamp = await rollup.read.getTimestampForSlot([slotsInEpoch]);
+    const cheatCodes = new EthCheatCodesWithState(t.ctx.aztecNodeConfig.l1RpcUrls);
+    try {
+      await cheatCodes.warp(Number(timestamp));
+    } catch (err) {
+      t.logger.debug('Warp failed, time already satisfied');
+    }
+
+    // Send and await a tx to make sure we mine a block for the warp to correctly progress.
+    await t.ctx.deployL1ContractsValues.publicClient.waitForTransactionReceipt({
+      hash: await t.ctx.deployL1ContractsValues.walletClient.sendTransaction({
+        to: t.baseAccount.address,
+        value: 1n,
+        account: t.baseAccount,
+      }),
+    });
+
+    // Set the system time in the node, only after we have warped the time and waited for a block
+    // Time is only set in the NEXT block
+    t.ctx.dateProvider.setTime(Number(timestamp) * 1000);
+
+    // create our network of nodes and submit txs into each of them
+    // the number of txs per node and the number of txs per rollup
+    // should be set so that the only way for rollups to be built
+    // is if the txs are successfully gossiped around the nodes.
+    const contexts: NodeContext[] = [];
+    t.logger.info('Creating nodes');
+    nodes = await createNodes(
+      t.ctx.aztecNodeConfig,
+      t.ctx.dateProvider,
+      t.bootstrapNodeEnr,
+      NUM_NODES,
+      BOOT_NODE_UDP_PORT,
+      t.prefilledPublicData,
+      DATA_DIR,
+      // To collect metrics - run in aztec-packages `docker compose --profile metrics up` and set COLLECT_METRICS=true
+      shouldCollectMetrics(),
+    );
+
+    // wait a bit for peers to discover each other
+    await sleep(4000);
+
+    t.logger.info('Submitting transactions');
+    for (const node of nodes) {
+      const context = await createPXEServiceAndSubmitTransactions(t.logger, node, NUM_TXS_PER_NODE, t.fundedAccount);
+      contexts.push(context);
+    }
+
+    t.logger.info('Waiting for transactions to be mined');
+    // now ensure that all txs were successfully mined
+    await Promise.all(
+      contexts.flatMap((context, i) =>
+        context.txs.map(async (tx, j) => {
+          t.logger.info(`Waiting for tx ${i}-${j}: ${await tx.getTxHash()} to be mined`);
+          return tx.wait({ timeout: WAIT_FOR_TX_TIMEOUT });
+        }),
+      ),
+    );
+    t.logger.info('All transactions mined');
+
+    // Gather signers from attestations downloaded from L1
+    const blockNumber = await contexts[0].txs[0].getReceipt().then(r => r.blockNumber!);
```

Comment:
dont really need from here downwards to perform the test. Its to do with the archiver storing the attestations for later

Reply:
Sure. Were thinking that this one should be taking over for the cheating one soon so was why they are same mostly.